django_lti_tool_provider/views.py

```diff
@@ -0,0 +1,142 @@
+from django_lti_tool_provider.signals import Signals
+
+
+_logger = logging.getLogger(__name__)
+
+
+class LTIView(View):
+    """ View handling LTI requests """
+    authentication_manager = None
+
+    PASS_TO_AUTHENTICATION_HOOK = {
+        'lis_person_sourcedid': 'username',
+        'lis_person_contact_email_primary': 'email',
+        'user_id': 'user_id'
+    }
+
+    SESSION_KEY = 'lti_parameters'
+
+    @csrf_exempt
+    def dispatch(self, *args, **kwargs):
+        if self.authentication_manager is None:
+            raise ImproperlyConfigured(u"AuthenticationManager is not set")
+
+        return super(LTIView, self).dispatch(*args, **kwargs)
+
+    def get(self, request, *args, **kwargs):
+        return self.process_request(request)
+
+    def post(self, request, *args, **kwargs):
+        return self.process_request(request)
+
+    def process_request(self, request):
+        if not request.user.is_authenticated():
+            try:
+                lti_parameters = self._get_lti_parameters_from_request(request)
+                lti_data = {
+                    hook_name: lti_parameters.get(lti_name, None)
+                    for lti_name, hook_name in self.PASS_TO_AUTHENTICATION_HOOK.items()
+                }
+            except oauth2.Error, e:
+                _logger.exception(u"Invalid LTI Request")
+                return HttpResponseBadRequest(u"Invalid LTI Request: " + e.message)
+
+            self.authentication_manager.authentication_hook(request, **lti_data)
+
+        if request.user.is_authenticated():
+            return self.process_authenticated_lti(request)
+        else:
+            return self.process_anonymous_lti(request)
+
+    @classmethod
+    def _get_lti_parameters_from_request(cls, request):
+        provider = DjangoToolProvider(settings.LTI_CLIENT_KEY, settings.LTI_CLIENT_SECRET, request.POST)
+        provider.valid_request(request)
+        return provider.to_params()
+
+    @classmethod
+    def _store_lti_parameters(cls, user, parameters):
+        """
+        Filters out OAuth parameters than stores LTI parameters into the DB, creating or updating record as needed
+        """
+        lti_params = {
+            key: value
+            for key, value in parameters.iteritems()
+            if 'oauth' not in key
+        }
+        custom_key = cls.authentication_manager.vary_by_key(lti_params)
+
+        # implicitly tested by test_views
+        if custom_key is None:
+            custom_key = ''
+
+        lti_user_data, created = LtiUserData.objects.get_or_create(user=user, custom_key=custom_key)
+        if not created:
+            _logger.debug(u"Replaced LTI parameters for user %s", user.username)
+        lti_user_data.edx_lti_parameters = lti_params
+        lti_user_data.save()
+        return lti_user_data
+
+    @classmethod
+    def register_authentication_manager(cls, manager):
+        """ Registers authentication manager """
+        cls.authentication_manager = manager
```

Comment:
You can provide additional keyword parameters that Django passes on to every view initialisation by passing them to `.as_view()` to avoid this being a global variable, but I don't care too much for Dalite.